crypto3/libs/zk/include/nil/crypto3/zk/snark/systems/plonk/placeholder/preprocessor.hpp

```diff
@@ -476,9 +477,13 @@ namespace nil {
                         const std::vector<std::size_t> &global_indices, // ordered global indices
                         const typename FieldType::value_type &omega,
                         const typename FieldType::value_type &delta,
-                        cycle_representation &permutation,
+                        const plonk_constraint_system<FieldType>& constraint_system,
+                        const plonk_table_description<FieldType>& table_description,
                         std::shared_ptr<math::evaluation_domain<FieldType>> domain
                     ) {
+                        // TODO: add std::vector<std::size_t> columns_with_copy_constraints;

```

Comment:
It is not about later object creation, but about the earlier destruction. In current code `permutation` object remains in memory until the end of `preprocess()` function which uselessly increases memory utilization by 15%

Reply:
btw @martun by the "proving itself" to you mean proof generation or verification? 

Asking because I do not observe memory peak on generating proof step
And verification steps fails on my sha256 example with the following logs
```
[2024-11-06 08:36:33.347964] [0x00000000074dafc0] [info]    Verifying proof...
[2024-11-06 08:43:12.802795] [0x00000000074dafc0] [info]    Verification failed because: F consolidated polynomial does not match.
```

Could you advise if this is expected now and which heavy inputs can I try to check the prover step now?

Reply:
will continue in DM, resolving this thread